core/services/ocr2/plugins/ocr2keeper/evm21/log_event_provider.go

```diff
@@ -0,0 +1,321 @@
+package evm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/i_keeper_registry_master_wrapper_2_1"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+// TODO: configurable values or based on block time
+const (
+	// logRetention is the amount of time to retain logs for.
+	// 5 minutes is the desired retention time for logs, but we add an extra 10 minutes buffer.
+	logRetention            = (time.Minute * 5) + (time.Minute * 10)
```

Comment:
🏄 For time values I think having the number precede the unit is more legible, e.g. `5 * time.Minute`